### PR TITLE
Add dladm_linkprop module

### PR DESCRIPTION
**Pull request by xen0l**
##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME
dladm_linkprop

##### ANSIBLE VERSION
```
2.3
```

##### SUMMARY

Module to manage network link interface properties on illumos/Solaris systems.

### CODE REVIEW

File: lib/ansible/modules/network/illumos/dladm_linkprop.py [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+dladm_linkprop: name=e1000g1 property=maxbw value=100M state=present
+
+# Set 'mtu' to 9000 on e1000g1
+dladm_linkprop: name=e1000g1 property=mtu value=9000
+
+# Reset 'mtu' property on e1000g1
+dladm_linkprop: name=e1000g1 property=mtu state=reset
+'''
+
+RETURN = '''
```
Comment (privateip) [inline lib/ansible/modules/network/illumos/dladm_linkprop.py]: RETURN doc string is required

File: lib/ansible/modules/network/illumos/dladm_linkprop.py [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+        result['changed'] = True
+
+    if out:
+        result['stdout'] = out
+    if err:
+        result['stderr'] = err
+
+    module.exit_json(**result)
+
+from ansible.module_utils.basic import *
```
Comment (privateip) [inline lib/ansible/modules/network/illumos/dladm_linkprop.py]: please use explicit imports

File: lib/ansible/modules/network/illumos/dladm_linkprop.py [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+main()
```
Comment (privateip) [inline lib/ansible/modules/network/illumos/dladm_linkprop.py]: please wrap in `IF` block

**Review by privateip — CHANGES_REQUESTED** [review]
* missing METADATA
* please consider adding unit tests

**Comment by xen0l** [comment]
ready_for_review

**Comment by xen0l** [comment]
@privateip I addressed you comments. As for unit tests, they will be shared among all modules to a certain degree (I am also speaking about modules, which are already included with Ansible). For that reason, I would like to add them in the next round if possible.

**Review by privateip — APPROVED** [review]

**Pull request merged**